### PR TITLE
Add warning for MP Metric API usage with Micrometer

### PR DESCRIPTION
**Pull request by kenfinnigan**
- Future versions will require use of SmallRye Metrics extension to use MP Metrics APIs

### CODE REVIEW

File: extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/mpmetrics/MicroprofileMetricsProcessor.java [hunk 1]
```diff
@@ -62,6 +62,25 @@ AdditionalBeanBuildItem registerBeanClasses() {
...
+            if (index.getAnnotations(annotation).size() > 0) {
+                mpMetricsPresent = true;
+                break;
+            }
+        }
+
+        if (mpMetricsPresent) {
+            log.warn("Use of MicroProfile Metrics annotations with the `quarkus-micrometer` extension in future Quarkus " +
+                    "versions will require using the `quarkus-smallrye-metrics` extension instead.");
+        }
```
Comment (ebullient) [inline extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/mpmetrics/MicroprofileMetricsProcessor.java:81]: We can't word the warning this way unless/until the smallrye-metrics and micrometer extensions _can_ be enabled at the same time.  What might be better is to warn that the MP Metrics API is being supported in a compatibility mode, which may be subject to change over time. (or something).

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicroprofileMetricsConfig.java [hunk 2]
```diff
@@ -14,15 +14,18 @@
...
      * <pre>
      * &lt;dependency&gt;
      *   &lt;groupId&gt;org.eclipse.microprofile.metrics&lt;/groupId&gt;
      *   &lt;artifactId&gt;microprofile-metrics-api&lt;/artifactId&gt;
      * &lt;/dependency&gt;
      * </pre>
+     * <p>
+     * Future versions of the micrometer extension will not include compatibility
+     * for MicroProfile Metrics. A future version will require the
+     * `quarkus-smallrye-metrics` extension for MicroProfile Metrics APIs with micrometer.
```
Comment (ebullient) [inline extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicroprofileMetricsConfig.java:28]: Same as above.

**Review by ebullient — CHANGES_REQUESTED** [review]

**Review by jmartisk — COMMENTED** [review]
Should we extend this beyond just annotations? This applies to `MetricRegistry` usage too AFAIK, so perhaps try to detect whether an application class injects a `MetricRegistry`?
Btw. Looks like one unwanted commit related to Qute strayed into this PR.

**Comment by kenfinnigan** [comment]
Do you know how we can find out whether it's the subject of an injection point?

**Comment by jmartisk** [comment]
Haven't tried it, but it seems like if you consume a `io.quarkus.arc.deployment.ValidationPhaseBuildItem` you can list all beans and their injection points.. perhaps that could do the job

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicroprofileMetricsConfig.java [hunk 3]
```diff
@@ -14,15 +14,19 @@
...
+     * support is enabled and the MicroProfile Metrics dependency is present:
      *
      * <pre>
      * &lt;dependency&gt;
      *   &lt;groupId&gt;org.eclipse.microprofile.metrics&lt;/groupId&gt;
      *   &lt;artifactId&gt;microprofile-metrics-api&lt;/artifactId&gt;
      * &lt;/dependency&gt;
      * </pre>
+     * <p>
+     * This application uses the MP Metrics API.
```
Comment (ebullient) [inline extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicroprofileMetricsConfig.java:26]: You can probably remove "This application uses the MP Metrics API." from the javadoc.

**Review by ebullient — APPROVED** [review]

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicroprofileMetricsConfig.java (hunk 3 above)
Comment (kenfinnigan) [reply extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicroprofileMetricsConfig.java]: Fixed now

**Comment by kenfinnigan** [comment]
How's this @jmartisk?

**Review by ebullient — APPROVED** [review]

**Comment by kenfinnigan** [comment]
@gsmet are you able to give this a quick look? I think Jan is PTO

**Review by dmlloyd — APPROVED** [review]

**Comment by kenfinnigan** [comment]
Thanks @dmlloyd

**Pull request merged**